api/writeAPIBlocking.go

```diff
@@ -80,18 +81,25 @@ func (w *writeAPIBlocking) write(ctx context.Context, line string) error {
 }
 
 func (w *writeAPIBlocking) WriteRecord(ctx context.Context, line ...string) error {
-	if len(line) > 0 {
-		var sb strings.Builder
-		for _, line := range line {
-			b := []byte(line)
-			b = append(b, 0xa)
-			if _, err := sb.Write(b); err != nil {
-				return err
-			}
+	if len(line) == 0 {
+		return nil
+	}
+	if len(line) == 1 {
+		ln := line[0]
+		if ln[len(ln)-1] != '\n' {
+			ln += "\n"
 		}
-		return w.write(ctx, sb.String())
+		return w.write(ctx, ln)
 	}
-	return nil
+	var sb strings.Builder
```

Comment:
a nit: this could be also optimized to avoid useless allocations of temporary buffers, it could Grow in advance to `len(line)+sum_of_line_lengths`

Reply:
`strings.Join(lines, "\n")` would probably be best, except for the need to add a terminal newline.  Lifting `[]bytes` out of the existing loop and expanding it at need based on line length could also work.  I didn't want to make that large a change, but if it's important enough I'll take a stab at it.

Reply:
yes `strings.Join` is already an optimized code and it would be the best option, though it would probably break the tests. The original goal herein (doc + simple optimization for 1 line) is already the necessary improvement.

The level and depth of optimizations is up to you ... it was just a nit from me. @vlastahajek is the approver herein.